src/main/java/seedu/address/logic/commands/ShowCommand.java

```diff
@@ -42,46 +46,71 @@
     public static final String MESSAGE_DUPLICATE_STUDENT_NAME =
             "This student needs to be specified using ID due to duplicate naming.";
 
+    private Index index;
     private Name name;
     private ID id;
     private Assessment assessment;
 
     /**
      * Constructor for a {@code ShowCommand}.
      */
-    public ShowCommand(Name name, ID id, Assessment assessment) {
+    public ShowCommand(Index index, Name name, ID id, Assessment assessment) {
+        setIndex(index);
         setName(name);
         setId(id);
         setAssessment(assessment);
     }
 
+    public ShowCommand(Index index) {
+        this(index, null, null, null);
+    }
+
     public ShowCommand(Name name) {
-        this(name, null, null);
+        this(null, name, null, null);
     }
 
     public ShowCommand(ID id) {
-        this(null, id, null);
+        this(null, null, id, null);
     }
 
     public ShowCommand(Assessment assessment) {
-        this(null, null, assessment);
+        this(null, null, null, assessment);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        return getAssessment().isEmpty() ? showStudent(model) : showAssessment(model);
+        return getIndex().isPresent() ? showStudentByIndex(model)
+                : getAssessment().isPresent() ? showAssessment(model)
+                : showStudentByPrefixes(model);
```

Comment:
```suggestion
        return getIndex().isPresent() 
                ? showStudentByIndex(model)
                : getAssessment().isPresent() 
                ? showAssessment(model)
                : showStudentByPrefixes(model);
```
For better readability